builders/build/substrate-api/polkadot-js-api.md

```diff
@@ -0,0 +1,301 @@
+
+And therefore, none of the information contained in the `api.{consts, query, tx}.<module>.<method>` endpoints are hard coded in the API. This allows parachains like Moonbeam to have custom endpoints through its pallets that can be directly accessed via polkadot.js API library.
+
+### State Queries {: #state-queries }
+
+This category of queries retrieves information related to the current state of the chain. These endpoints are generally of the form, `api.query.<module>.<method>`, where the module and method decorations are generated through metadata. 
+
+Here is a code sample for retrieving basic account information given its address:
+
+```javascript
+// Initialize the API as in previous sections
+...
+
+// Define wallet address
+const ADDR = 'insert-moonbeam-network-wallet-address';
+
+// Retrieve the last timestamp
+const now = await api.query.timestamp.now();
+
+// Retrieve the account balance & nonce via the system module
+const { nonce, data: balance } = await api.query.system.account(ADDR);
+
+console.log(`${now}: balance of ${balance.free} and a nonce of ${nonce}`);
+```
+
+### RPC Queries {: #rpc-queries }
+
+The RPC calls provide the backbone for the transmission of data to and from the node. This means that all API endpoints such as `api.query`, `api.tx` or `api.derive` just wrap RPC calls, providing information in the encoded format as expected by the node.
+
+The `api.rpc` interface follows the a similar format to `api.query`, for instance:
+
+```javascript
+// Retrieve the chain name
+const chain = await api.rpc.system.chain();
+
+// Retrieve the latest header
+const lastHeader = await api.rpc.chain.getHeader();
+
+// Log the information
+console.log(`${chain}: last block #${lastHeader.number} has hash ${lastHeader.hash}`);
+```
+
+### Query Subscriptions {: #query-subscriptions }
+
+The RPCs lend themselves to using subscriptions. We can adapt the previous example to start using subscriptions to listen to new blocks.
+
+```javascript
+// Subscribe to the new headers
+await api.rpc.chain.subscribeNewHeads((lastHeader) => {
+  console.log(`${chain}: last block #${lastHeader.number} has hash ${lastHeader.hash}`);
+});
+```
+
+The general pattern for `api.rpc.subscribe*` functions is to pass a callback into the subscription function, and this will be triggered on each new entry as they are imported. 
+
+Other calls under `api.query.*` can be modifed in a similar fashion to use subscription, including calls that have parameters. Here is an example to subscribe to balance changes in an account:
+
+```javascript
+// Subscribe to balance changes for our account
+const unsub = await api.query.system.account(ADDR, ({ nonce, data: balance }) => {
+  console.log(`free balance is ${balance.free} with ${balance.reserved} reserved and a nonce of ${nonce}`);
+});
+```
+
+## Keyrings and Transactions {: #keyrings-and-transactions }
```

Comment:
Probably put keyrings in its own section just below provider? What do you think? Keyring is not really tied to txs only